homeassistant/config_entries.py

```diff
@@ -817,13 +817,25 @@ def async_get_options_flow(config_entry: ConfigEntry) -> "OptionsFlow":
         raise data_entry_flow.UnknownHandler
 
     @callback
-    def _abort_if_unique_id_configured(self) -> None:
+    def _abort_if_unique_id_configured(
+        self, update_entry: Dict[Any, Any] = None
+    ) -> None:
         """Abort if the unique ID is already configured."""
         if self.unique_id is None:
             return
 
-        if self.unique_id in self._async_current_ids():
-            raise data_entry_flow.AbortFlow("already_configured")
+        for entry in self._async_current_entries():
+            if entry.unique_id == self.unique_id:
+                if (
+                    self.hass is not None
+                    and update_entry is not None
+                    and not update_entry.items() <= entry.data.items()
+                ):
+                    data = (entry.data or {}).copy()
```

Comment:
`entry.data` always exists.

You could do `{**entry.data, **update_entry}` too.

Reply:
`entry.data` -> Yes MyPy things differently about that.